core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -93,6 +91,9 @@ class SparkHadoopUtil extends Logging {
         if (key.startsWith("spark.hadoop.")) {
           hadoopConf.set(key.substring("spark.hadoop.".length), value)
         }
+        if (key.startsWith("spark.yarn.")) {
```

Comment:
This is simple but feels hacky. You're setting non-Hadoop props in the Hadoop config and relying on Hadoop doing some variable expansion.